test/lib/repo_discovery_test.rb

```diff
@@ -58,13 +58,15 @@ def test_docker_with_v1_search_no_proxy
       crawled = []
       found = []
       to_follow = [base_url]
-      search = 'busybox'
+      upstream_credentials_and_search = {
+        search: 'busybox'
+      }
 
       RestClient::Request.expects(:execute)
-        .with({ method: :get, url: base_url.to_s + "v1/search?q=#{search}", headers: {:accept => :json} })
+        .with({ method: :get, url: base_url.to_s + "v1/search?q=#{upstream_credentials_and_search[:search]}", headers: {:accept => :json} })
```

Comment:
I'd just hardcode this because it's a better guarantee. If something modifies the hash in place you can silently end up with `q=` instead of `q=busybox`. Previously with a string that could also happen, but it was less likely.
```suggestion
        .with({ method: :get, url: base_url.to_s + "v1/search?q=busybox", headers: {:accept => :json} })
```